src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -211,6 +211,19 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
             }
         }
 
+        public override BoundNode Visit(BoundNode node)
+        {
+            if (_inExpressionLambda &&
+                !(node is BoundConversion) &&
+                node is BoundExpression expr &&
+                expr.Type is TypeSymbol type &&
+                type.IsByRefLikeType)
+            {
+                Error(ErrorCode.ERR_FeatureNotValidInExpressionTree, node, "ref struct");
```

Comment:
I think it would be fine to just put the type's name here.  We don't want unlocalized strings like this in diagnostics.  We do have a localized feature name if you really want this text.

Reply:
@agocke's suggestion in https://github.com/dotnet/roslyn/pull/30871#discussion_r230208523 seems reasonable